fixtures/flight/src/Button.js

```diff
@@ -6,20 +6,22 @@ export default function Button({action, children}) {
   const [isPending, setIsPending] = React.useState(false);
 
   return (
-    <button
-      disabled={isPending}
-      onClick={async () => {
-        setIsPending(true);
-        try {
-          const result = await action();
-          console.log(result);
-        } catch (error) {
-          console.error(error);
-        } finally {
-          setIsPending(false);
-        }
-      }}>
-      {children}
-    </button>
+    <form>
```

Comment:
Buttons that use this technique need to be paired with a `<form>` because that's what gets submitted to navigation in this model. Unfortunately, you can't use them free-standing. It's extra unfortunate since forms can't be nested so you can't just necessarily always include a form in a button abstraction neither.

Reply:
I assume that this only applies if the button `type` attribute is not explicitly set to `”button”`, right?

Reply:
But why would we wrap it in a form instead of changing the type in this standalone scenario?

Reply:
Currently it'll silently fail if there's no form. Probably need to have a contextual nesting warning for this.

Reply:
Got it, I didn’t see that you also changed `onClick` to `formAction`.

Reply:
Since React general knows whether we're in a form or not, I guess we could have a feature that just creates a dummy form somewhere that can be referenced by orphan buttons.